digdag-tests/src/test/java/acceptance/RequireIT.java

```diff
@@ -185,6 +203,159 @@ public void testIgnoreProjectIdParam()
+        );
+        assertThat(startStatus.errUtf8(), startStatus.code(), is(0));
+    }
+
+    @Test
+    public void testRerunOnParam()
+            throws Exception
+    {
+        // Create new project
+        CommandStatus initStatus = main("init",
+                "-c", config.toString(),
+                projectDir.toString());
+        assertThat(initStatus.errUtf8(), initStatus.code(), is(0));
+
+        Path childOutFile = projectDir.resolve("child.out").toAbsolutePath().normalize();
+        prepareForChildWF(childOutFile);
+        copyResource("acceptance/require/parent_rerun_on.dig", projectDir.resolve("parent.dig"));
+        copyResource("acceptance/require/child_rerun_on.dig", projectDir.resolve("child.dig"));
+
+        // Push the project
+        CommandStatus pushStatus = main("push",
+                "--project", projectDir.toString(),
+                "require",
+                "-c", config.toString(),
+                "-e", server.endpoint(),
+                "-r", "4711");
+        assertThat(pushStatus.errUtf8(), pushStatus.code(), is(0));
+
+        // test for rerun_on: none
+        {
+            String sessionTime = "2020-05-28 12:34:01";
+            RestSessionAttemptCollection childAttempts = testRerunOnParam(sessionTime, "none", false);
+            assertThat("Number of child's attempt must be one. (== require> skip the call)", childAttempts.getAttempts().size(), is(1));
+        }
+
+        // test for rerun_on: all. require> kick child always.
+        {
+            String sessionTime = "2020-05-28 12:34:11";
+            RestSessionAttemptCollection childAttempts = testRerunOnParam(sessionTime, "all", false);
+            assertThat("Number of child's attempt must be two. (== require> kick child)", childAttempts.getAttempts().size(), is(2));
+        }
+
+        // test for rerun_on: failed. previous child attempt failed
+        {
+            String sessionTime = "2020-05-28 12:34:21";
+            RestSessionAttemptCollection childAttempts = testRerunOnParam(sessionTime, "failed", true);
+            assertThat("Number of child's attempt must be two. (== require> kick child)", childAttempts.getAttempts().size(), is(2));
+        }
+
+        // test for rerun_on: failed. previous child attempt succeeded
+        {
+            String sessionTime = "2020-05-28 12:34:31";
+            RestSessionAttemptCollection childAttempts = testRerunOnParam(sessionTime, "failed", false);
+            assertThat("Number of child's attempt must be one. (== require> kick)", childAttempts.getAttempts().size(), is(1));
+        }
+    }
+
+    private RestSessionAttemptCollection testRerunOnParam(String sessionTime, String rerunOn, boolean previousChildRunFail) throws InterruptedException
+    {
+        //Start a child
+        String childFailParam = previousChildRunFail ? "yes" : "no";
+        CommandStatus childStatus = startAndWait(true,"require", "child", "--session", sessionTime,
```

Comment:
I'm wondering why this method call always passes `ignoreAttemptFailure=true` (sorry, a bit too difficult for me 😅.)

Could I ask you the followings?
1. Add a comment for each case about why target attempt is expected to fail
2. If possible, make this method call pass `attemptShouldFail` or something deterministic instead of ` ignoreAttemptFailure`

Reply:
After having read all codes in `testRerunOnParam()` I understand all things. So probably you can ignore the above comment from me.

Reply:
I will add some additional comments.

Reply:
Done. Thanks!